src/synapse/workload.py

```diff
@@ -859,39 +860,28 @@ def get_environment(charm_state: CharmState) -> typing.Dict[str, str]:
     return environment
 
 
-def generate_nginx_config(container: ops.Container, main_unit_address: str) -> None:
+def generate_nginx_config(main_unit_address: str) -> None:
     """Generate NGINX configuration based on templates.
 
     1. Copy template files as configuration files to be used.
     2. Run sed command to replace string main-unit in configuration files.
     3. Reload NGINX.
 
     Args:
-        container: Container of the charm.
         main_unit_address: Main unit address to be used in configuration.
     """
-    container.exec(
-        [
-            "cp",
-            "/etc/nginx/main_location.conf.template",
-            "/etc/nginx/main_location.conf",
-        ],
-    ).wait()
-    container.exec(
-        ["sed", "-i", f"s/main-unit/{main_unit_address}/g", "/etc/nginx/main_location.conf"],
-    ).wait()
-    container.exec(
-        [
-            "cp",
-            "/etc/nginx/abuse_report_location.conf.template",
-            "/etc/nginx/abuse_report_location.conf",
-        ],
-    ).wait()
-    container.exec(
-        [
-            "sed",
-            "-i",
-            f"s/main-unit/{main_unit_address}/g",
-            "/etc/nginx/abuse_report_location.conf",
-        ],
-    ).wait()
+    file_loader = FileSystemLoader("./nginx_rock/etc", followlinks=True)
+    env = Environment(loader=file_loader)
+
+    # List of templates and their corresponding output files
+    templates = [
+        ("main_location.conf.j2", "main_location.conf"),
+        ("abuse_report_location.conf.j2", "buse_report_location.conf"),
+    ]
+
+    for template_name, output_file in templates:
+        template = env.get_template(template_name)
+        output = template.render(main_unit_address=main_unit_address)
+
+        with open(output_file, "w", encoding="utf-8") as f:
+            f.write(output)
```

Comment:
I think you are writing these files to the charm container rather than the workload container. You probably need to use something like https://ops.readthedocs.io/en/latest/#ops.Container.push to push the content into the workload container.